tests/src/PatternAssertion/CarouselPatternAssert.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\PatternAssertion;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the carousel pattern.
+ */
+class CarouselPatternAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions(string $variant): array {
+    return [
+      'items' => [
+        [$this, 'assertItems'],
+      ],
+      'settings' => [
+        [$this, 'assertSettings'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $crawler = new Crawler($html);
+
+    $this->assertElementExists('.carousel .carousel-inner', $crawler);
+  }
+
+  /**
+   * Asserts the carousel pattern settings.
+   *
+   * @param array[] $expected
+   *   The expected settings.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The crawler.
+   */
+  protected function assertSettings(array $expected, Crawler $crawler): void {
```

Comment:
I think we should provide the default values for all the settings and always test for them, as we pass them directly down to the template and to form the "api layer" we need to make sure their values don't change.